x/mongo/driver/operation/hello.go

```diff
@@ -113,6 +121,430 @@ func (h *Hello) Result(addr address.Address) description.Server {
 	return description.NewServer(addr, bson.Raw(h.res))
 }
 
+const (
+	// FaaS environment variable names
+	envVarAWSExecutionEnv        = "AWS_EXECUTION_ENV"
+	envVarAWSLambdaRuntimeAPI    = "AWS_LAMBDA_RUNTIME_API"
+	envVarFunctionsWorkerRuntime = "FUNCTIONS_WORKER_RUNTIME"
+	envVarKService               = "K_SERVICE"
+	envVarFunctionName           = "FUNCTION_NAME"
+	envVarVercel                 = "VERCEL"
+)
+
+const (
+	// FaaS environment variable names
+	envVarAWSRegion                   = "AWS_REGION"
+	envVarAWSLambdaFunctionMemorySize = "AWS_LAMBDA_FUNCTION_MEMORY_SIZE"
+	envVarFunctionMemoryMB            = "FUNCTION_MEMORY_MB"
+	envVarFunctionTimeoutSec          = "FUNCTION_TIMEOUT_SEC"
+	envVarFunctionRegion              = "FUNCTION_REGION"
+	envVarVercelURL                   = "VERCEL_URL"
+	envVarVercelRegion                = "VERCEL_REGION"
+)
+
+const (
+	// FaaS environment names used by the client
+	envNameAWSLambda = "aws.lambda"
+	envNameAzureFunc = "azure.func"
+	envNameGCPFunc   = "gcp.func"
+	envNameVercel    = "vercel"
+)
+
+// getFaasEnvName parses the FaaS environment variable name and returns the
+// corresponding name used by the client. If none of the variables or variables
+// for multiple names are populated the client.env value MUST be entirely
+// omitted.
+func getFaasEnvName() string {
+	envVars := []string{
+		envVarAWSExecutionEnv,
+		envVarAWSLambdaRuntimeAPI,
+		envVarFunctionsWorkerRuntime,
+		envVarKService,
+		envVarFunctionName,
+		envVarVercel,
+	}
+
+	// If none of the variables are populated the client.env value MUST be
+	// entirely omitted.
+	names := make(map[string]struct{})
+
+	for _, envVar := range envVars {
+		if os.Getenv(envVar) == "" {
+			continue
+		}
+
+		var name string
+
+		switch envVar {
+		case envVarAWSExecutionEnv, envVarAWSLambdaRuntimeAPI:
+			name = envNameAWSLambda
+		case envVarFunctionsWorkerRuntime:
+			name = envNameAzureFunc
+		case envVarKService, envVarFunctionName:
+			name = envNameGCPFunc
+		case envVarVercel:
+			name = envNameVercel
+		}
+
+		names[name] = struct{}{}
+		if len(names) > 1 {
```

Comment:
It seems that we only need a string instead of a map. We can check whether the string is empty.

Reply:
Both "AWS_EXECUTION_ENV" and "AWS_LAMBDA_RUNTIME_API" will result in the name "aws.lambda". It's technically possible for both variables to be populated. In this case if we just checked to see if the "name" string is populated, then "getFaasEnvName" would return an empty string which is unexpected behavior. That is, the "names" set exists in the case of aliases.

[Here](https://github.com/mongodb/specifications/blob/master/source/mongodb-handshake/handshake.rst#clientenv) are the specifications.